### PR TITLE
 Add test with werimock in servive external api

### PR DESCRIPTION
**Pull request by luismiguel0529**
## Summary
- Add test service external api
## Trello Card
https://trello.com/c/vdNwQaTF/43-mockear-respuestas-con-wiremock

### CODE REVIEW

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+        wireMockServer
+                .stubFor(get(urlEqualTo(String.format(url, isbnFail)))
+                        .willReturn(aResponse().withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                                .withStatus(200)
+                                .withBodyFile("{}")));
+    }
+
+    @Test
+    @DisplayName("Test,When find a book ,it retunr a book")
```
Comment (joredudiaz) [inline src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: `Test, when search a book ,it returns a book`

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    void whenFindBookByIsbnThenReturnBook() {
+        BookDTO bookDTO = openLibraryService.findInfoBook(isbnSuccess);
+        Assert.assertEquals("0385472579", bookDTO.getIsbn());
+        Assert.assertEquals("Zen speaks", bookDTO.getTitle());
+        Assert.assertEquals("shouts of nothingness", bookDTO.getSubtitle());
+        Assert.assertEquals("1994", bookDTO.getPublishDate());
+    }
+
+    @Test
+    @DisplayName("Test,When find a book ,it retunr a not found")
```
Comment (joredudiaz) [inline src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: `Test, when search a book ,it returns not found`

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import wolox.training.exception.BookNotFoundException;
+import wolox.training.models.dto.BookDTO;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
```
Comment (joredudiaz) [inline src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: no utilices wildcards (*)

**Review by joredudiaz — CHANGES_REQUESTED** [review]

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java (hunk 3 above)
Comment (luismiguel0529) [reply src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: corrected

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java (hunk 2 above)
Comment (luismiguel0529) [reply src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: corrected

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java (hunk 1 above)
Comment (luismiguel0529) [reply src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: corrected

**Review by joredudiaz — DISMISSED** [review]

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+public class OpenLibraryServiceTest {
+
+    @Autowired
+    private OpenLibraryService openLibraryService;
+
+    private WireMockServer wireMockServer;
+    private static final String isbnSuccess = "0385472579";
+    private static final String isbnFail = "2";
+
+    @BeforeEach
```
Comment (facucompe) [inline src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: BeforeAll

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Test
+    @DisplayName("Test, when search a book ,it returns not found")
+    void whenFIndBookByIsbnThenReturnNotFound() {
+        Assertions.assertThrows(BookNotFoundException.class, () -> openLibraryService.findInfoBook(isbnFail));
+    }
+
+
+    @AfterEach
```
Comment (facucompe) [inline src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: AfterAll

**Review by facucompe — COMMENTED** [review]
That small correction

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java (hunk 4 above)
Comment (luismiguel0529) [reply src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: corrected

File: src/test/java/wolox/training/service/OpenLibraryServiceTest.java (hunk 5 above)
Comment (luismiguel0529) [reply src/test/java/wolox/training/service/OpenLibraryServiceTest.java]: corrected

**Review by facucompe — APPROVED** [review]

**Review by joredudiaz — APPROVED** [review]

**Pull request merged**